CMakeLists.txt

```diff
@@ -153,14 +159,15 @@ if(APPLE)
     set(XCODE_DEVELOPMENT_TEAM "" CACHE STRING "Xcode development team ID")
     set(PRODUCTBUILD_IDENTITY_NAME "" CACHE STRING "productbuild identity name")
     set(PRODUCTBUILD_KEYCHAIN_PATH "" CACHE FILEPATH "pkgbuild keychain file")
-    if(IOS)
+    if(APPLE_VAR_OS)
         set(XCODE_PROVISIONING_PROFILE_SPECIFIER "" CACHE STRING "Xcode provisioning profile specifier")
         set(MOLTENVK_SDK "$ENV{VULKAN_SDK}/../MoltenVK" CACHE STRING "MoltenVK directory inside the Vulkan SDK")
     endif()
 
     # Deployment
-    set(CMAKE_OSX_DEPLOYMENT_TARGET "10.11" CACHE STRING "macOS Deployment Target")
-    if(IOS)
+    if(APPLE_MAC_OS)
```

Comment:
Is this really necessary? `CMAKE_OSX_DEPLOYMENT_TARGET` only applies for macOS so there is no harm in always setting it.

Reply:
That's very misleading on CMake's behalf:
> Despite the OSX part in the variable name(s) they apply also to other SDKs than macOS like iOS, tvOS, visionOS, or watchOS.
-- [CMake documentation](https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_DEPLOYMENT_TARGET.html)

I think that never bit us, because in the only other case (iOS), up until now, we're setting the target via `CMAKE_XCODE_ATTRIBUTE_IPHONEOS_DEPLOYMENT_TARGET` in the lines below, which seems to override it.

Reply:
Thanks for the clarification. Is there a `CMAKE_XCODE_ATTRIBUTE_MACOS_DEPLOYMENT_TARGET`? If so perhaps we should use that. Oh! Do these `XCODE_ATTRIBUTES` work for other generators on macOS? If not, we should continue to use `CMAKE_OSX_DEPLOYMENT_TARGET`.

Reply:
Yeah, I've learned that myself just now.
Yes, let's stick to `CMAKE_OSX_DEPLOYMENT_TARGET`, since it seems to be the default CMake way to do things.